### DISCUSSION TITLE
TH3 Testing

### DISCUSSION THREAD

**Discussion by emindeniz99** [tursodatabase/libsql#1479] [General]
I saw that th3 is not open source but it is beneficial test for SQLite. Are there any attempt to include it in libsql?

https://news.ycombinator.com/item?id=40693802

**Comment by emindeniz99** [comment]
https://news.ycombinator.com/item?id=33100113

**Comment by Jiehong** [comment]
I'm not related to the project, but I was wondering about the testing strategy, and if libSQL actually benefits from the "Deterministic Simulation Testing" of Turso.

Coming up with an answer to this might be tricky, apart from running TH3 on both and compare, or to have a different testing suite also used as comparison.
